Microsoft.Toolkit.Uwp.UI.Controls/CameraPreview/CameraPreview.cs

```diff
@@ -0,0 +1,171 @@
+
+            _frameSourceGroupCombo.SelectionChanged += FrameSourceGroupCombo_SelectionChanged;
+
+            await InitFrameSourcesAsync();
+        }
+
+        private async Task InitFrameSourcesAsync()
+        {
+            var frameSourceGroups = await FrameSourceGroupsHelper.GetAllAvailableFrameSourceGroupsAsync();
+
+            if (frameSourceGroups?.Count > 0)
+            {
+                _frameSourceGroupCombo.ItemsSource = frameSourceGroups;
+            }
+            else
+            {
+                _frameSourceGroupCombo.ItemsSource = new List<object> { new { DisplayName = "No camera sources found." } };
+            }
+            _frameSourceGroupCombo.SelectedIndex = 0;
+        }
+
+        private void SetMediaPlayerSource()
+        {
+            var frameSource = _cameraHelper?.FrameSource;
+            if (frameSource != null)
+            {
+                if (_mediaPlayer == null)
+                {
+                    _mediaPlayer = new MediaPlayer
+                    {
+                        AutoPlay = true,
+                        RealTimePlayback = true
+                    };
+                }
+
+                _mediaPlayer.Source = MediaSource.CreateFromMediaFrameSource(frameSource);
+                _mediaPlayerElementControl.SetMediaPlayer(_mediaPlayer);
+            }
+        }
+
+        private void CameraHelper_VideoFrameArrived(object sender, VideoFrameEventArgs e)
+        {
+            EventHandler<VideoFrame> vfHandler = VideoFrameArrived;
+            vfHandler?.Invoke(sender, e.VideoFrame);
+
+            EventHandler<SoftwareBitmap> sbHandler = SoftwareBitmapArrived;
+            sbHandler?.Invoke(sender, e.SoftwareBitmap);
+        }
+
+        private async void FrameSourceGroupCombo_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            var selectedGroup = _frameSourceGroupCombo.SelectedItem as MediaFrameSourceGroup;
+            if (selectedGroup != null)
+            {
+                if (_cameraHelper == null)
+                {
+                    _cameraHelper = new CameraHelper();
+
+                    // Subscribe to the video frame as they arrive
+                    _cameraHelper.VideoFrameArrived += CameraHelper_VideoFrameArrived;
+                }
+
+                var result = await _cameraHelper.InitializeAndStartCaptureAsync(selectedGroup);
+
+                if (result.Status)
+                {
+                    SetMediaPlayerSource();
+                }
+                else
+                {
+                    _mediaPlayerElementControl.SetMediaPlayer(null);
+                }
+
+                _videoPreviewErrorTextBlock.Text = result.Message;
+                _videoPreviewErrorTextBlock.Visibility = result.Status ? Visibility.Collapsed : Visibility.Visible;
+                _videoPreviewText.Visibility = !result.Status ? Visibility.Collapsed : Visibility.Visible;
```

Comment:
nit: instead of checking if(!result.Status == true), don't negate result.Status and invert your logic